docs/DeveloperGuide.md

```diff
@@ -4,9 +4,457 @@
 
```

Comment:
Great job on including very detailed descriptions of each component of your code.